### PR TITLE
Adds Error class for when Zuora returns invalid request data

### PR DESCRIPTION
**Pull request by jwg2s**
**Developer Checklist**
- [x] I considered alternative approaches and chose the simplest one.
- [x] I wrote unit/acceptance specs for these changes that test all possible conditions (success, failure, bad input, etc).
- [x] I wrote unit/acceptance specs for code/functionality these changes may have affected.
- [x] I can't think of any other code/functionality this may have affected that I didn't test or account for.
- [x] I've ensured any necessary data migrations have been run.
- [x] I've ensured other blocking code has been deployed (e.g. if part of a larger feature).
- [x] We are able to track any changes to application performance or utilization introduced by this PR.

**Deploying is blocked by:**
- [x] Nothing :shipit:

### CODE REVIEW

File: lib/zuora/errors.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    class MissingRequiredValue < StandardError
+      attr_reader :response
+
+      def initialize(msg = nil, response = nil)
+        @response = response
+        super(msg)
+      end
+    end
+  end
+end
```
Comment (env) [inline lib/zuora/errors.rb]: Maybe need newline at EOF?

File: lib/zuora/errors.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Zuora
+  module Errors
+    class MissingRequiredValue < StandardError
+      attr_reader :response
+
+      def initialize(msg = nil, response = nil)
```
Comment (env) [inline lib/zuora/errors.rb]: `nil` is default...default. so don't need to specify here

File: lib/zuora/client.rb [hunk 3]
```diff
@@ -55,7 +55,11 @@ def request!(body)
         request.body = body.to_xml
       end
 
-      Zuora::Response.new(raw_response)
+      response = Zuora::Response.new(raw_response)
+
+      Zuora::Response.handle_errors(response.to_h)
```
Comment (env) [inline lib/zuora/client.rb]: would it make sense to make this an instance method?
````ruby
response = Zuora::Response.new(raw_response)
response.handle_errors
response
```

File: lib/zuora/errors.rb (hunk 1 above)
Comment (jwg2s) [inline lib/zuora/errors.rb]: :+1:

**Comment by jwg2s** [comment]
Reviewed by @env

**Pull request merged**